6.0.0rc1 pytest.mark.xfail mypy error: No overload variant

<!--
Thanks for submitting an issue!

Here's a quick checklist for what to provide:
-->

- [ ] a detailed description of the bug or suggestion
- [ ] output of `pip list` from the virtual environment you are using
- [ ] pytest and operating system versions
- [x] minimal example if possible

using example from https://docs.pytest.org/en/documentation-restructure/how-to/skipping.html#raises-parameter

```
import pytest

@pytest.mark.xfail(raises=RuntimeError)
def test_function():
    ...
```

running mypy

```
$ mypy ~/pytest-mypy.py
C:\Users\simon\pytest-mypy.py:3: error: No overload variant of "__call__" of "_XfailMarkDecorator" matches argument type "Type[RuntimeError]"
C:\Users\simon\pytest-mypy.py:3: note: Possible overload variant:
C:\Users\simon\pytest-mypy.py:3: note:     def __call__(self, condition: Union[str, bool] = ..., *conditions: Union[str, bool], reason: str = ..., run: bool = ..., raises: Union[BaseException, Tuple[BaseException, ...]] = ..., strict: bool = ...) -> MarkDecorator
C:\Users\simon\pytest-mypy.py:3: note:     <1 more non-matching overload not shown>
Found 1 error in 1 file (checked 1 source file)
```

should the type annotation for raise in _XfailMarkDecorator be raises: Union[Type[BaseException], Tuple[Type[BaseException], ...]] = ...,

xref https://github.com/pandas-dev/pandas/issues/35261#issuecomment-657597413

--- a/src/_pytest/mark/structures.py
+++ b/src/_pytest/mark/structures.py
@@ -28,6 +28,9 @@ from _pytest.config import Config
 from _pytest.outcomes import fail
 from _pytest.warning_types import PytestUnknownMarkWarning
 
+if TYPE_CHECKING:
+    from typing import Type
+
 
 EMPTY_PARAMETERSET_OPTION = "empty_parameter_set_mark"
 
@@ -413,7 +416,9 @@ if TYPE_CHECKING:
             *conditions: Union[str, bool],
             reason: str = ...,
             run: bool = ...,
-            raises: Union[BaseException, Tuple[BaseException, ...]] = ...,
+            raises: Union[
+                "Type[BaseException]", Tuple["Type[BaseException]", ...]
+            ] = ...,
             strict: bool = ...
         ) -> MarkDecorator:
             raise NotImplementedError()

